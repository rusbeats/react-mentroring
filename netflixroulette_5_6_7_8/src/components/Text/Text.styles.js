import styled from 'styled-components';

export const StyledText = styled.div`
    color: ${props => props.color};
    cursor: ${props => props.cursor};
    pointer-events: ${props => props.pointerEvents};
`;
