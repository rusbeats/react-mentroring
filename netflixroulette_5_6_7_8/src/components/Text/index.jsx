import React from 'react';
import PropTypes from 'prop-types';
import { StyledText } from './Text.styles';

const Text = ({ color, cursor, pointerEvents, children, onClick }) => {
    return (
        <StyledText onClick={onClick} color={color} cursor={cursor} pointerEvents={pointerEvents}>
            {children}
        </StyledText>
    );
};

StyledText.displayName = 'Text';

Text.propTypes = {
    color: PropTypes.string,
    cursor: PropTypes.string,
    pointerEvents: PropTypes.string,
    children: PropTypes.node,
    onClick: PropTypes.func
};

export default Text;
