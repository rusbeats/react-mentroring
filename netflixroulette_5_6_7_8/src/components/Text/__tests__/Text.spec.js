import React from 'react';
import { shallow } from 'enzyme';
import Text from '../index';

const PROPS = {
    color: '#000000',
    cursor: 'pointer',
    pointerEvents: 'none',
    children: 'Text for render',
    onClick: jest.fn()
};

describe('Text', () => {
    it('should render correctly (snapshot)', () => {
        const component = shallow(
            <Text color={PROPS.color} cursor={PROPS.cursor} pointerEvents={PROPS.pointerEvents} onClick={PROPS.onClick}>
                {PROPS.children}
            </Text>
        );
        expect(component).toMatchSnapshot();
    });
});
