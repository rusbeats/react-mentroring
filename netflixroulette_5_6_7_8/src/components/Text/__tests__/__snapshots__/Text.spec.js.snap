// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render correctly (snapshot) 1`] = `
<Text
  color="#000000"
  cursor="pointer"
  onClick={[MockFunction]}
  pointerEvents="none"
>
  Text for render
</Text>
`;
